scripts/bootstrap.sh

```diff
@@ -18,14 +20,21 @@ if test "$PWD" != "$BACKEND_SERVER_DIR"; then
   exit 1
 fi
 
-if test -z "$1"
+if test -z "$BACKEND"
 then
   echo "Please specify your backend when running this file. For example:"
   echo "$0 my_backend";
   exit 1
 fi
 
-echo "BACKEND_JUST=$1" > .env
+# put in .env to ensure `just` has access to it
+echo "BACKEND=$BACKEND" > .env
+
+# also put it in the default profile so everything else has access to it
+mkdir -p /etc/opensafely/profile.d
+BACKEND_PROFILE=/etc/opensafely/profile.d/backend.sh
+grep -q "$BACKEND" $BACKEND_PROFILE || printf "readonly BACKEND=%s\nexport BACKEND" "$BACKEND" >> $BACKEND_PROFILE
+ln -s $BACKEND_PROFILE /etc/profile.d/backend.sh
```

Comment:
I'd be inclined to call this `opensafely.sh`, as we may end up sticking other things than backend in it?

Reply:
Hmm, but this is dynamically generated.

Yeah, just an idea, up to you

Reply:
actually we already have https://github.com/opensafely-core/backend-server/blob/main/etc/opensafely/profile but I was going to keep them separate, as backend-server should never change (once created) & other stuff might do? not a strong opinion, though, can fold into the older file if you rather?

Reply:
oh, also, because I need to write the envvar from bootstrap, it needs to use echo/printf (not even `cat << EOF`), which makes it unwieldy - the other stuff in `opensafely/profile` is static so just a flat file

Reply:
going to merge this as-is, we can always do more tidying later

Reply:
:+1: good call